### PR TITLE
peering between different versions

### PR DESCRIPTION
**Pull request by roosnic1**
WPC-112. This PR changes the way how to add a new peer. Instead of accessing the API of the new peer the editor sends a query to it's own API which will then delegate the request to the other API. This way we can better handle changes in the GraphQL Schema.

### CODE REVIEW

File: packages/editor/src/client/panel/peerEditPanel.tsx [hunk 1]
```diff
@@ -1,25 +1,15 @@
...
 
 import {
   PeerListDocument,
   useCreatePeerMutation,
   usePeerQuery,
   useUpdatePeerMutation,
-  PeerProfileDocument,
-  FullPeerProfileFragment
+  FullPeerProfileFragment,
+  useRemotePeerPrfoileQuery
```
Comment (penk85) [inline packages/editor/src/client/panel/peerEditPanel.tsx:12]: Little typo here: profile

Comment (roosnic1) [reply packages/editor/src/client/panel/peerEditPanel.tsx:12]: Good catch. Thank you :)

File: packages/editor/src/client/api/index.ts [hunk 2]
```diff
@@ -1340,6 +1342,12 @@ export type Query = {
 };
 
 
+export type QueryRemotePeerProfileArgs = {
+  hostURL: Scalars['String'];
```
Comment (camilleory) [inline packages/editor/src/client/api/index.ts:1345]: what's the difference between using `Scalars['String'];` or simply `String`?

File: packages/editor/src/client/api/index.ts [hunk 3]
```diff
@@ -6118,6 +6147,42 @@ export function usePeerProfileLazyQuery(baseOptions?: Apollo.LazyQueryHookOption
...
+ *      hostURL: // value for 'hostURL'
+ *      token: // value for 'token'
+ *   },
+ * });
+ */
+export function useRemotePeerPrfoileQuery(baseOptions: Apollo.QueryHookOptions<RemotePeerPrfoileQuery, RemotePeerPrfoileQueryVariables>) {
+        const options = {...defaultOptions, ...baseOptions}
+        return Apollo.useQuery<RemotePeerPrfoileQuery, RemotePeerPrfoileQueryVariables>(RemotePeerPrfoileDocument, options);
+      }
+export function useRemotePeerPrfoileLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<RemotePeerPrfoileQuery, RemotePeerPrfoileQueryVariables>) {
```
Comment (camilleory) [inline packages/editor/src/client/api/index.ts]: What does this lazy query does ?

**Review by camilleory — APPROVED** [review]
Found a spelling mistake and asked a couple of questions

File: packages/editor/src/client/api/index.ts [hunk 4]
```diff
@@ -3196,6 +3208,20 @@ export type PeerProfileQuery = (
   ) }
 );
 
+export type RemotePeerPrfoileQueryVariables = Exact<{
```
[the remark this answers is not in the document]
Comment (roosnic1) [reply packages/editor/src/client/api/index.ts]: already fixed :)

File: packages/editor/src/client/api/index.ts (hunk 2 above)
Comment (roosnic1) [reply packages/editor/src/client/api/index.ts:1345]: We use https://www.graphql-code-generator.com to generate this file out of the `*.graphql` files. So you don't have to review it in the PR. I think scalars belongs to something graphql specific.

File: packages/editor/src/client/api/index.ts (hunk 3 above)
Comment (roosnic1) [reply packages/editor/src/client/api/index.ts]: graphql-codegen creates two kind of queries. One to execute directly and another one that can be executed on a click. https://www.apollographql.com/docs/react/data/queries/#manual-execution-with-uselazyquery

**Review by bishoymelek — APPROVED** [review]
🎉 🎉

**Review by penk85 — APPROVED** [review]

**Review by lakeside763 — APPROVED** [review]

**Pull request merged**